src/API/Google/AdsAssetGroup.php

```diff
@@ -190,11 +205,80 @@ protected function listing_group_delete_operations(): array {
 	}
 
 	/**
-	 * Return a temporary resource name for the campaign.
+	 * Return a temporary resource name for the asset group.
 	 *
 	 * @return string
 	 */
 	protected function temporary_resource_name() {
 		return ResourceNames::forAssetGroup( $this->options->get_ads_id(), self::TEMPORARY_ID );
 	}
+
+	/**
+	 * Get Asset Groups for a specific campaign.
+	 *
+	 * @since x.x.x
+	 *
+	 * @param int  $campaign_id The campaign ID.
+	 * @param bool $include_assets Whether to include the assets in the response.
+	 *
+	 * @return array The asset groups for the campaign.
+	 */
+	public function get_asset_groups_by_campaign_id( int $campaign_id, bool $include_assets = true ): array {
```

Comment:
❓  In regards `$include_assets` param. Is there any case we don't want to get the assets?

Reply:
Currently, we are listing only one asset group with the assets included however in the future we could list all asset groups for a specific campaign without requiring the assets so the user can choose which asset group to load the data. 

Also, I added the param because it was not adding complexity to the function and was making the function more flexible.